### DISCUSSION TITLE
SPH metrics incorrectly mapped to PVOutput?

### DISCUSSION THREAD

**Discussion by robupham** [johanmeijer/grott#174] [General]
Hi,

Firstly, thanks for this project and for making it public - it's really appreciated.

I run an SPH system with battery (currently SPH3000, about to be upgraded to SPH6000). I'm using the invtype=sph parameter in grott.ini and appear to be picking up data correctly. However, when I view my uploaded data at PVOutput it seems that something's not quite right.

Example here: https://pvoutput.org/intraday.jsp?id=106917&sid=93558&dt=20220925

Some observations:
The "Energy" metric seems incorrect. It grows throughout the day - but crucially grows during the night when I know I'm not generating any PV Output.
The "Power" metric similarly reports power apparently generated during the night. I know this isn't solar output - I suspect it's counting discharge from my battery and reporting that. (in the example above, I know my battery ran out of charge at just after 5am).
The "Average" column makes no sense to me.
"Temperature", ""Energy Used" and "Power Used" are empty.

I'd be happy to provide further info to aid in troubleshooting - but wondered if this was a known issue.

**Comment by johanmeijer** [comment]
I can understand that the pvoutput values for the hybrid inverters (SPH) and off grid inverters (SPF) are not good. I have not tested this and Grott might sent the wrong values to pvoutput. 

For the SPH grott sends the output power for powerfield (pvoutput field V2). , I think that is the power that is sent from the inverter to the grid. At night that probably will be power from the battery. At daytime this proably will be the power that is "left" after battery charging (so power generated by the solar panels - power used for charging the batteries). But I am not sure. 

For energy field (V1) eacttoday (Today generate energy) is used. That does not seems bad to me, but I am not sure. 

You can disable the sending of the energy field by specifying pvdisv1 = True. in grott.ini. The power field is then only used and the energy is calculated by pvoutput. This gives also a more logical average overview (although I think the values show are not wrong but looking strange because of detailed information, see also closed issue #52)  

The temperature should be the outside temperature I think. That information is not in the inverter data. On request I build an option to use the temperature field for the inverter temperature. You can use this by specifying pvtemp = True in grott.ini. 

We have to look for the right values for the SPH. Maybe you can find them at the growatt server side or look at this document (input regs): https://www.scribd.com/document/489508943/Growatt-Inverter-Modbus-RTU-Protocol-II-V1-20-English 

In one of the next releases of grott (V3+) I will make it more easy to select the values your self that you want to be sent to pvoutput. For now we can try to find the right values for SPH and SPF and change that in Grott.
